magento/devdocs#: Cloud. Add VCL for the sitemap.xml and robots.txt files (#5330)
magento/devdocs: guides/v2.2/cloud/trouble/robots-sitemap.md

With {{site.data.var.ece}}, you can only write to specific directories, such as `var`, `pub/media`, `pub/static`, or `app/etc`. When you generate the `sitemap.xml` file using the Admin panel, you must specify the `/media/` path.

> You do not have to generate a `robots.txt` because it is generated on demand and stored in the database. It does not create a file, but you can view the content in your browser with the `<domain.your.project>/robots.txt` file.

Reviewer: Changed to passive?

Author: **done**

Reviewer: I don't see a change here. Revert.

Author: Should I revert this sentence to 
```
You do not have to generate a `robots.txt` because it generates on demand and stores the contents in the database. It does not create a file, but you can view the content in your browser with the `<domain.your.project>/robots.txt` file.
```
?

Thank you!

Reviewer: I am sorry for the confusion I caused. I suggest:

```suggestion
You do not have to generate a `robots.txt` file because it generates the `robots.txt` content on demand and stores it in the database. You can view the content in your browser with the `<domain.your.project>/robots.txt` link.
```

Author: **fixed**

This requires ECE-Tools version 2002.0.12 and later with an updated `.magento.app.yaml` file. See an example of these rules in the [magento-cloud repository](https://github.com/magento/magento-cloud/blob/master/.magento.app.yaml#L43-L49).

> ## To generate a `sitemap.xml` file in version 2.2 and later

Reviewer: Procedure titles are h4 at the moment.

Author: **done**

Reviewer: I don't see a change.

Author: > I don't see a change.

Could you please clarify the expected result here? Thank you!

Author: @hguthrie, `##` means `h4` or I am wrong?

Thanks

Author: **fixed**

1. Click **Save & Generate**. The new site map becomes available in the _Site Map_ grid.
1. Click the path in the _Link for Google_ column.

> ## To add content to `robots.txt` file

Reviewer: h4

Author: **done**

Reviewer: I don't see a change.

Author: **fixed**

  "content": "if ( req.url.path ~ \"^/?sitemap.xml$\" ) { set req.url = \"/media/sitemap.xml\"; }"
}
```

> See the example below to understand how to rewrite the path for `robots.txt` and `sitemap.xml` to `/media/robots.txt` and `/media/sitemap.xml` accordingly in the single snippet:

Reviewer: Consider:

```suggestion
The following example demonstrates how to rewrite the path for `robots.txt` and `sitemap.xml` to `/media/robots.txt` and `/media/sitemap.xml`
```

Author: **done**

```json
{
  "name": "sitemaprobots_rewrite",
  "dynamic": "0",
  "type": "recv",
  "priority": "90",
  "content": "if ( req.url.path ~ \"^/?sitemap.xml$\" ) { set req.url = \"/media/sitemap.xml\"; } else if (req.url.path ~ \"^/?robots.txt$\") { set req.url = \"/media/robots.txt\";}"
}
```

> ### To use a Fastly VCL snippet for particular domain redirect

Reviewer: ```suggestion
#### To use a Fastly VCL snippet for particular domain redirect:
```

Author: **done**

Create a `pub/media/domain_robots.txt` file, where the domain is `domain.com` and use the next VCL snippet:

```
{
  "name": "domain_robots",
  "dynamic": "0",
  "type": "recv",
  "priority": "90",
  "content": "if ( req.url.path == \"/robots.txt\" ) { if ( req.http.host ~ \"(domain).com$\" ) { set req.url = \"/media/\" re.group.1 \"_robots.txt\"; }}"
}
```

> VCL will take care of routing `http://domain.com/robots.txt` and present the correct file.

Reviewer: remove "will"

"takes care of" is very subjective. It is possible you could say " The VCL snippet routes and presents the ...file."

Author: **done**

> To configure redirect for `robots.txt` and `sitemap.xml` in the single snippet, create `pub/media/domain_robots.txt` and `pub/media/domain_sitemap.xml`, where the domain is `domain.com` and use the next VCL snippet:

Reviewer: ```suggestion
To configure a redirect for `robots.txt` and `sitemap.xml` in a single snippet, create `pub/media/domain_robots.txt` and `pub/media/domain_sitemap.xml` files, where the domain is `domain.com` and use the next VCL snippet:
```

Author: **done**

```
{
  "name": "domain_sitemaprobots",
  "dynamic": "0",
  "type": "recv",
  "priority": "90",
  "content": "if ( req.url.path == \"/robots.txt\" ) { if ( req.http.host ~ \"(domain).com$\" ) { set req.url = \"/media/\" re.group.1 \"_robots.txt\"; }} else if ( req.url.path == \"/sitemap.xml\" ) { if ( req.http.host ~ \"(domain).com$\" ) {  set req.url = \"/media/\" re.group.1 \"_sitemap.xml\"; }}"
}
```

> The admin config for the `sitemap` you will need to specify that the file actually is in `pub/media/` rather than `/`.

Reviewer: ```suggestion
In the `sitemap` admin config, you must specify the location of the file using `pub/media/` rather than `/`.
```
Might be nice to explain why.

Author: **done**
